proto/api.go

```diff
@@ -308,14 +307,139 @@ func (sr *ReverseScanResponse) Verify(req Request) error {
 	return nil
 }
 
+// GetInner returns the Request contained in the `oneof`.
+func (ru BatchRequest_RequestUnion) GetInner() Request {
```

Comment:
This is ugly because we're not really using `oneof` in the right way here; we're using it to express a field holding an interface type. `onlyone` generates the switch statement for us, but it does a worse job; the return value is `interface{}`. In this case, there's still a benefit to writing the switch statement ourselves: we can give this getter the proper return type and avoid type casts in various places.

That said, it might be a reasonable feature request to gogo/protobuf to add support for generating exactly this sort of switch statement.

Finally, the current version of gogo/protobuf forces us to choose between `oneof` and `onlyone`, and considers `onlyone` deprecated.